### PR TITLE
Fix repo refresh

### PR DESCRIPTION
**Pull request by tdm4**
Hi, I found an issue where it takes 2 runs of puppet to fix the package installation. Now the packages correctly require Class[::apt::update] so they install correctly the first time.

### CODE REVIEW

**Comment by tdm4** [comment]
@mmoll I'm not sure why it's showing my older commits from #190  .. but as you can see, only 3 files changed in this PR. Do you know how I'd fix the commits? Do I have to squash them again?

**Comment by mmoll** [comment]
I guess you did branch off from a branch that did hold the old history from before your commits were squashed when merging #190. In general I recommend to always have master/develop of a fork in sync with the upstrema repo, but never do commits to it, then use topic branches for PRs to the upstream repo and when opening new PRs, re-sync master/develop from upstream and branch off a new topic branch from that. Anyway, from a mayflower/puppet-php perspective it's perfectly fine to squash all these commits together into one when merging.

File: manifests/dev.pp [hunk 1]
```diff
@@ -27,7 +27,12 @@
   }
 
   if $::operatingsystem == 'Ubuntu' {
-    ensure_packages(["${php::globals::package_prefix}xml"])
+    if ( ! defined(Package["${php::globals::package_prefix}xml"])) {
```
Comment (mmoll) [inline manifests/dev.pp]: is there a way around that? using `defined` would really be a last resort measurement

Comment (mmoll) [inline manifests/dev.pp]: using `ensure_packages` from stdlib might be in order here.

**Comment by tdm4** [comment]
@mmoll Hmm.. I tried squashing these to one commit but I am getting an error: Cannot 'squash' without a previous commit
Any ideas? The command I ran was: git rebase -i HEAD~7 since my fork is 7 commits ahead.

File: manifests/dev.pp (hunk 1 above)
Comment (tdm4) [inline manifests/dev.pp]: You can't use `ensure_packages` with a require.

**Comment by tdm4** [comment]
I wonder if I'm better off re-forking mayflower-php master and redoing this PR? I can't seem to squash the commits.

**Comment by tdm4** [comment]
@mmoll I think ensure_packages might take a hash of resources. Going to see if I can clean this up.

**Comment by mmoll** [comment]
you can always grab the current mayflower master and force push it into the master branch of your fork and then cherry-pick your commit into another branch and force push into the fix_repo_refresh of your fork.

**Comment by tdm4** [comment]
OK I've sorted the ugly if ( ! defined(Package[...])) stuff. Re-checking stdlib, it does take resources in a hash after the package list which is ace. So I've amended the PR to suit.

**Comment by mmoll** [comment]
merged, thanks @tdm4!

**Pull request merged**